specification/intelligentdataexperiences/resource-manager/Microsoft.AadCustomSecurityAttributesDiagnosticSettings/preview/2023-03-01-preview/intelligentdataexperiences.json

```diff
@@ -0,0 +1,486 @@
+            "description": "Error. The request to create diagnostic setting has failed.",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          }
+        }
+      },
+      "delete": {
+        "tags": ["DiagnosticSettings"],
+        "operationId": "DiagnosticSettings_Delete",
+        "description": "Deletes existing diagnostic setting for AadCustomSecurityAttributesDiagnosticSettings.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error. The request to delete named diagnostic setting has failed.",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          },
+          "200": { "description": "Successful request to remove a diagnostic setting" },
+          "204": { "description": "Successful request to remove a diagnostic setting" }
+        },
+        "x-ms-examples": {
+          "Deletes the diagnostic setting": { "$ref": "./examples/deleteDiagnosticSetting.json" }
+        }
+      }
+    },
+    "/providers/Microsoft.AadCustomSecurityAttributesDiagnosticSettings/diagnosticSettingsCategories": {
+      "get": {
+        "tags": ["DiagnosticSettingsCategories"],
+        "operationId": "DiagnosticSettingsCategory_List",
+        "description": "Lists the diagnostic settings categories for AadCustomSecurityAttributesDiagnosticSettings.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error. The request to view diagnostic setting categories has failed.",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          },
+          "200": {
+            "description": "Successful request to get list of diagnostic setting categories",
+            "schema": { "$ref": "#/definitions/DiagnosticSettingsCategoryResourceCollection" }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the diagnostic setting categories": { "$ref": "./examples/listDiagnosticSettingsCategories.json" }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ProxyOnlyResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "A proxy only azure resource object.",
+      "type": "object"
+    },
+    "RetentionPolicy": {
+      "properties": {
+        "enabled": {
```

Comment:
>enabled [](http://example.com/codeflow?start=9&length=7)

It is better to use enum (with x-ms-enum) instead of booleans wherever possible. Foe example, here, you can use:

The "state" enum with values "enabled", "disabled", which makes it extensible in future.

See LintDIff errors for more details.